### DISCUSSION TITLE
Feature Idea: CLI `--cleanup=false` option to retain temp config files after execution

### DISCUSSION THREAD

**Discussion by aaronsteers** [meltano/meltano#6441] [Ideas]
We currently have `--dump` options to unload config files. Another approach would be to offer a `--cleanup=false` option that performs the following functions:

1. Run all operations as usual.
2. Skip cleanup/purge of interim config files.
3. Print a message at end of execution which files were _not_ deleted so the user knows (1) where the files reside and (2) how to properly perform the necessary manual cleanup after execution.

----------

- Q: To the team and community: Would this feature be helpful?

In theory, this would be helpful if combined with `--dry-run` or `--dry` options, but those may or may not be compatible with these options in a first release if they otherwise would be a complete "noop" today.

### Reason not to build

1. In theory, a similar operation is available today using the `--dump` option.
2. This creates something of a security risk if not used responsibly by users, since the left-behind config files will in many cases contain creds and other secrets.

**Comment by edgarrmondragon** [comment]
@aaronsteers I really like the idea. Some thoughts:

- what does the team think of a simple boolean flag instead of a specified value? i.e. `--no-cleanup`
- I kinda wish we could extend `meltano remove` to make the plugin type and name optional, and so have a `--temp-files` flag to remove all catalog, config or other temporary files for all plugins

**Comment by rafalkrupinski** [comment]
that would be great!
